auth_proxy/proxy/flask.py

```diff
@@ -36,37 +36,53 @@ class FlaskClient(Client):
         'Procedure',
     ]
 
+    SECURITY_ARG_NAME = '_security'
+
     def __init__(self, url, orig):
         self.url = url
         self.orig = orig
 
     def request(self):
         """ @inherit
         """
-        args = list(self.orig.args.items())
 
-        # We don't care what the client thinks it should be able to see
-        args = [arg for arg in args if arg[0] != '_security']
+        stripped_args, stripped_headers = self._get_secured_args_and_headers()
+
+        return {
+            'headers': stripped_headers,
+            'method': self.orig.method,
+            'url': self._create_url(stripped_args),
+            'body': self.orig.data,
+        }
+
+    def _get_secured_args_and_headers(self):
+        """
+        Enforce security on the request by removing existing security parameters and 
+        adding our own based on scope/patient authorizations. Also remove headers 
+        not in the allowed list.
+        :return: list, dict : The modified request arguments and headers.
+        """
+
+        original_args = list(self.orig.args.items())
+
+        # Strip existing security arguments so we can apply our own.
+        stripped_args = [arg for arg in original_args if arg[0] != self.SECURITY_ARG_NAME]
 
-        # Update the query params
+        # Add security based on the requested path.
         path = self.orig.view_args.get('path').split('/')
+
         if len(path) == 1:
-            args.append(('_security', self._scope_security()))
-            args.append(('_security', self._patient_security()))
+            stripped_args.append((self.SECURITY_ARG_NAME, self._get_scope_security_label()))
```

Comment:
Sorry I didn't catch this before it got merged.

My understanding of Josh's request was that these lines be changed to something more along the lines of

```
args = self._enforce_security_labels(original_args)
```

so that you'd have a single, functional, method to override instead of multiple smaller methods that mutate state.

You wouldn't need to create the `_get_secured_args_and_headers` method, and could keep all that logic in request like before. Then, `UnsecureFlaskClient` can make `_enforce_security_labels` a straight passthrough function, and doesn't need to override request. Like I said before, I think that overriding requests obscures the intention of this class. UnsecureFlaskClient doesn't exist to do requests differently, it exists to disable security.

Reply:
Thanks for the followup note Erik! I agree: this reflects the approach I was advocating. @mtmcduffie perhaps you might submit a quick follow-up? (Sorry for my rush to merge earlier today; that's what it gets for multitasking.)

Incidentally: after working more in the Google paradigm of "reviewers create reviews to approve changes, and developers merge their own PRSC when they're reviewing *and* the developers is happy with any cleanup / final tweaks", I think it might be good for us to adopt in S4S. Probably we can discuss this over email ;)